Failure with test_download_ftp_file_properly_handles_socket_error

While testing 4.3rc1:
```
_____________________ test_download_ftp_file_properly_handles_socket_error _____________________

    @pytest.mark.remote_data
    def test_download_ftp_file_properly_handles_socket_error():
        faulty_url = "ftp://anonymous:mail%40astropy.org@nonexisting/pub/products/iers/finals2000A.all"
        with pytest.raises(urllib.error.URLError) as excinfo:
            download_file(faulty_url)
        errmsg = excinfo.exconly()
        found_msg = False
        possible_msgs = ['Name or service not known',
                         'nodename nor servname provided, or not known',
                         'getaddrinfo failed',
                         'Temporary failure in name resolution']
        for cur_msg in possible_msgs:
            if cur_msg in errmsg:
                found_msg = True
                break
>       assert found_msg, f'Got {errmsg}, expected one of these: {",".join(possible_msgs)}'
E       AssertionError: Got urllib.error.URLError: <urlopen error [Errno -5] No address associated with hostname>, expected one of these: Name or service not known,nodename nor servname provided, or not known,getaddrinfo failed,Temporary failure in name resolution
E       assert False

.pyenv/versions/astropy43/lib/python3.9/site-packages/astropy/utils/tests/test_data.py:2197: AssertionError
```

--- a/astropy/utils/tests/test_data.py
+++ b/astropy/utils/tests/test_data.py
@@ -2189,7 +2189,8 @@ def test_download_ftp_file_properly_handles_socket_error():
     possible_msgs = ['Name or service not known',
                      'nodename nor servname provided, or not known',
                      'getaddrinfo failed',
-                     'Temporary failure in name resolution']
+                     'Temporary failure in name resolution',
+                     'No address associated with hostname']
     for cur_msg in possible_msgs:
         if cur_msg in errmsg:
             found_msg = True

